File: spring-cloud-old/CubicCustomerService-Ch03/src/main/java/com/cubic/microservices/customer_service/domain/Customer.java
package com.cubic.microservices.customer_service.domain;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "CUSTOMER")
public class Customer {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "CUST_ID", nullable = false)
	private int id;
	
	@Column(name = "CUST_NO", nullable = false)
	private String customerNo;

	@Column(name = "NAME", nullable = false)
	private String name;
	
	@Column(name = "EMAIL", nullable = false)
	private String email;

	@Column(name = "ADDRESS", nullable = false)
	private String address;

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getCustomerNo() {
		return customerNo;
	}

	public void setCustomerNo(String customerNo) {
		this.customerNo = customerNo;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getAddress() {
		return address;
	}

	public void setAddress(String address) {
		this.address = address;
	}

	@Override
	public String toString() {
		return "Customer [id=" + id + ", customerNo=" + customerNo + ", name=" + name + ", email=" + email
				+ ", address=" + address + "]";
	}

}

File: microservices-angular/ch02/src/app/customer.ts
import { Account } from './account';

export class Customer {
    id: number;
    customerNo: string;
    name: string;
    email: string;
    address: string;
    accountList: Account[];

    // cloneAAA(): Customer {
    //     let customer = new Customer ( );
    //     customer.id = this.id;
    //     customer.customerNo = this.customerNo;
    //     customer.name = this.name;
    //     customer.email = this.email;
    //     customer.address = this.address;
    //     return customer;
    // }

    // test() : string {
    //     return "abc";
    // }
}
